src/base/bittorrent/torrentinfo.cpp

```diff
@@ -456,6 +483,31 @@ void TorrentInfo::stripRootFolder()
     m_nativeInfo->remap_files(files);
 }
 
+void TorrentInfo::addRootFolder()
+{
+    const QString rootFolder = name();
+    Q_ASSERT(!rootFolder.isEmpty());
+
+    const std::string rootPrefix = Utils::Fs::toNativePath(rootFolder + QLatin1Char {'/'}).toStdString();
+    lt::file_storage files = m_nativeInfo->files();
+    files.set_name(rootFolder.toStdString());
+    for (int i = 0; i < files.num_files(); ++i)
+        files.rename_file(lt::file_index_t {i}, rootPrefix + files.file_path(lt::file_index_t {i}));
+    m_nativeInfo->remap_files(files);
+}
+
+TorrentContentLayout TorrentInfo::defaultContentLayout() const
+{
+    QStringList origFilePaths;
```

Comment:
Usually I would choose `QVector<QString>`. However in Qt6, `QList` has become an alias to `QVector` (not really but more closer), so at least you should add `origFilePaths.reserve();` here.